cmd-script.c

```diff
@@ -120,6 +120,16 @@ int command_script(int argc, char *argv[], struct opts *opts)
 		return -1;
 	}
 
+	if (opts->record) {
+		char *script_file = opts->script_file;
+		opts->script_file = NULL;
```

Comment:
To apply in-script record option, you need to call `parse_script_opt()` before making it NULL.